app/models/certificate.rb

```diff
@@ -127,77 +127,15 @@ def revoke!
       return false
     end
 
-    self.class.update_registry_crl
-    self.class.reload_apache
+    self.class.update_crl
     self
   end
 
   class << self
     def update_crl
-      update_id_crl
-      update_registry_crl
-      reload_apache
-    end
-
-    def update_id_crl
-      STDOUT << "#{Time.zone.now.utc} - Updating ID CRL\n" unless Rails.env.test?
-
-      _out, _err, _st = Open3.capture3("
-        mkdir -p #{ENV['crl_dir']}/crl-id-temp
-        cd #{ENV['crl_dir']}/crl-id-temp
-
-        wget https://sk.ee/crls/esteid/esteid2007.crl
-        wget https://sk.ee/crls/juur/crl.crl
-        wget https://sk.ee/crls/eeccrca/eeccrca.crl
-        wget https://sk.ee/repository/crls/esteid2011.crl
-
-        openssl crl -in esteid2007.crl -out esteid2007.crl -inform DER
-        openssl crl -in crl.crl -out crl.crl -inform DER
-        openssl crl -in eeccrca.crl -out eeccrca.crl -inform DER
-        openssl crl -in esteid2011.crl -out esteid2011.crl -inform DER
-
-        ln -s crl.crl `openssl crl -hash -noout -in crl.crl`.r0
-        ln -s esteid2007.crl `openssl crl -hash -noout -in esteid2007.crl`.r0
-        ln -s eeccrca.crl `openssl crl -hash -noout -in eeccrca.crl`.r0
-        ln -s esteid2011.crl `openssl crl -hash -noout -in esteid2011.crl`.r0
-
-        rm -rf #{ENV['crl_dir']}/*.crl #{ENV['crl_dir']}/*.r0
-
-        mv #{ENV['crl_dir']}/crl-id-temp/* #{ENV['crl_dir']}
-
-        rm -rf #{ENV['crl_dir']}/crl-id-temp
-      ")
-
-      STDOUT << "#{Time.zone.now.utc} - ID CRL updated\n" unless Rails.env.test?
-    end
-
-    def update_registry_crl
-      STDOUT << "#{Time.zone.now.utc} - Updating registry CRL\n" unless Rails.env.test?
-
-      _out, _err, _st = Open3.capture3("
-        mkdir -p #{ENV['crl_dir']}/crl-temp
-        cd #{ENV['crl_dir']}/crl-temp
-
-        openssl ca -config #{ENV['openssl_config_path']} -keyfile #{ENV['ca_key_path']} -cert \
-        #{ENV['ca_cert_path']} -gencrl -out #{ENV['crl_dir']}/crl-temp/crl.pem -key \
-        '#{ENV['ca_key_password']}' -batch
-
-        ln -s crl.pem `openssl crl -hash -noout -in crl.pem`.r1
-
-        rm -rf #{ENV['crl_dir']}/*.pem #{ENV['crl_dir']}/*.r1
-
-        mv #{ENV['crl_dir']}/crl-temp/* #{ENV['crl_dir']}
-
-        rm -rf #{ENV['crl_dir']}/crl-temp
-      ")
-
-      STDOUT << "#{Time.zone.now.utc} - Registry CRL updated\n" unless Rails.env.test?
-    end
-
-    def reload_apache
-      STDOUT << "#{Time.zone.now.utc} - Reloading apache\n" unless Rails.env.test?
-      _out, _err, _st = Open3.capture3("sudo /etc/init.d/apache2 reload")
-      STDOUT << "#{Time.zone.now.utc} - Apache reloaded\n" unless Rails.env.test?
+      STDOUT << "#{Time.zone.now.utc} - Running crlupdater\n" unless Rails.env.test?
+      system "#{ENV['crl_updater_path']}"
```

Comment:
If someone can configure our application to point to malicious script or even produce one at our server we have other problems i don't think this would introduce any new vectors for attack.

Even if someone is able to modify the script it's still run with the privileges of registry application and only reload/restart of webserver to reload the crl is ran with elevated privileges so there is no new privilege escalation scenario introduced.

CRL reloading isn't used with all applications for example registry admin portal does not use any CRL's so requiring that env var to be set for all applications does not seem reasonable, our current approach of figaro does not allow us to define per application required vars as far as i know, so i have solved that in our deployment tool rather then application itself.